Bottom: 56602790aed630ddb581fe3d7e4ff0b9807d7130
Top:    cead7d356b816bc45f29bb402bd4e5a583a048a1
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 14:07:56 +0100

Attempt to debug islandness after inserting hole bug


---

diff --git a/src/action.c b/src/action.c
index b32493c..7bdda3e 100644
--- a/src/action.c
+++ b/src/action.c
@@ -1504,12 +1504,19 @@ NotifyMode (void)
 		   */
 		  SaveUndoSerialNumber ();
 		  Flags = ((PolygonType *)Crosshair.AttachedObject.Ptr2)->Flags;
-		  PolyToPoursOnLayer (PCB->Data, (LayerType *)Crosshair.AttachedObject.Ptr1,
-				      result, Flags);
+#warning FIXME: Removing a pour should force a re-check of islanding!
+                  /* For now, do this before PolyToPours() which re-inits the
+                   * clipping for the pour(s) affected (?). Ideally RemoveObject
+                   * should fix any island response its self. Anyway.. removing
+                   * the old thing first saves us doing twice the work by
+                   * recomputing islands when it gets deleted.
+                   */
 		  RemoveObject (POUR_TYPE,
 				Crosshair.AttachedObject.Ptr1,
 				Crosshair.AttachedObject.Ptr2,
 				Crosshair.AttachedObject.Ptr3);
+		  PolyToPoursOnLayer (PCB->Data, (LayerType *)Crosshair.AttachedObject.Ptr1,
+				      result, Flags);
 		  RestoreUndoSerialNumber ();
 		  IncrementUndoSerialNumber ();
 		  Draw ();
